File: Transportation/src/main/java/com/example/transportation/entity/Offer.java
package com.example.transportation.entity;

import com.example.transportation.enums.OfferState;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.ZonedDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Offer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String information;
    private Double price;
    private ZonedDateTime creatingTime = ZonedDateTime.now();


    @ManyToOne()
    private Transporter transporter;

    @ManyToOne()
    private Delivery delivery;


    @Enumerated(EnumType.STRING)
    private OfferState state;



}

File: TransportationFront/src/app/api/models/Offer.ts
import {Delivery} from "@api/models/Delivery";
import {Transporter} from "@api/models/Transporter";
import {OfferState} from "@api/models/enums/OfferState";

export interface Offer{
  id:number;
  information:string;
  price:number;

  delivery:Delivery;
  transporter:Transporter;
  state:OfferState;
}
